torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -341,6 +342,25 @@ def get_default_x86_inductor_quantization_config(
     return quantization_config
 
 
+@functools.lru_cache
+def get_x86_inductor_linear_dynamic_fp16_config():
+    """
+    For linear_dynamic_fp16. The name may be confusing.
+    The op's behavior is fp32_input * (fp16_weight -> to_fp32) -> fp32_output.
+    """
+    weight_quantization_spec = QuantizationSpec(
+        dtype=torch.float16,
+        observer_or_fake_quant_ctr=PlaceholderObserver,
+    )
+    quantization_config = QuantizationConfig(
+        None,  # input_quantization_spec
```

Comment:
we don't need `input_quantization_spec` for dynamic quant?

Reply:
The op is called `dynamic` but only weight is converted to fp16. I have checked the kernel implementation and found that computation is like `x_fp32 * (w_fp16 -> to_fp32) -> y_fp32`. `x` is not converted to fp16 at all.

Reply:
I see, OK..